donate/settings/environment.py

```diff
@@ -0,0 +1,95 @@
+                   'client-analytics.sandbox.braintreegateway.com', 'api.braintreegateway.com',
+                   'client-analytics.braintreegateway.com', '*.braintree-api.com', 'www.paypal.com',
+                   'www.google-analytics.com', 'https://www.mozilla.org/en-US/newsletter/',
+                   'https://sentry.prod.mozaws.net')
+
+# We set defaults for values that aren't security related
+# to the least permissive setting. For security related values,
+# we rely on it being explicitly set (no default values) so that
+# we error out first.
+env = environ.Env(
+    ALLOWED_HOSTS=(list, '*'),
+    AUTH_PASSWORD_VALIDATORS=(list, []),
+    AWS_ACCESS_KEY_ID=(str, ''),
+    AWS_LOCATION=(str, ''),
+    AWS_REGION=(str, ''),
+    AWS_S3_CUSTOM_DOMAIN=(str, ''),
+    AWS_SECRET_ACCESS_KEY=(str, ''),
+    AWS_STORAGE_BUCKET_NAME=(str, ''),
+    BASKET_API_ROOT_URL=(str, ''),
+    BASKET_SQS_QUEUE_URL=(str, ''),
+    BRAINTREE_MERCHANT_ACCOUNTS=(dict, {}),
+    BRAINTREE_MERCHANT_ACCOUNTS_PAYPAL_MICRO=(dict, {}),
+    BRAINTREE_MERCHANT_ID=(str, ''),
+    BRAINTREE_PLANS=(dict, {}),
+    BRAINTREE_PRIVATE_KEY=(str, ''),
+    BRAINTREE_PUBLIC_KEY=(str, ''),
+    BRAINTREE_TOKENIZATION_KEY=(str, ''),
+    BRAINTREE_USE_SANDBOX=(bool, True),
+    CONTENT_TYPE_NO_SNIFF=(bool, True),
+    CSRF_COOKIE_SECURE=(bool, False),
+    CSP_BASE_URI=(tuple, CSP_BASE_URI),
+    CSP_CONNECT_SRC=(tuple, CSP_CONNECT_SRC),
+    CSP_DEFAULT_SRC=(tuple, CSP_DEFAULT_SRC),
+    CSP_FONT_SRC=(tuple, CSP_FONT_SRC),
+    CSP_FORM_ACTION=(tuple, CSP_DEFAULT_SRC),
+    CSP_FRAME_ANCESTORS=(tuple, CSP_DEFAULT_SRC),
+    CSP_FRAME_SRC=(tuple, CSP_FRAME_SRC),
+    CSP_IMG_SRC=(tuple, CSP_IMG_SRC),
+    CSP_MEDIA_SRC=(tuple, CSP_DEFAULT_SRC),
+    CSP_OBJECT_SRC=(tuple, CSP_DEFAULT_SRC),
+    CSP_REPORT_URI=(tuple, None),
+    CSP_SANDBOX=(tuple, CSP_DEFAULT_SRC),
+    CSP_SCRIPT_SRC=(tuple, CSP_SCRIPT_SRC),
+    CSP_STYLE_SRC=(tuple, CSP_STYLE_SRC),
+    CSP_WORKER_SRC=(tuple, CSP_WORKER_SRC),
+    DATABASE_URL=(str, 'postgres://donate:mozilla@postgres:5432/donate'),
+    DEBUG=(bool, False),
+    DJANGO_LOG_LEVEL=(str, 'INFO'),
+    DJANGO_SECRET_KEY=(str, None),
+    DOMAIN_REDIRECT_MIDDLEWARE_ENABLED=(bool, False),
+    HEROKU_APP_NAME=(str, ''),
+    HEROKU_RELEASE_VERSION=(str, None),
+    RANDOM_SEED=(int, None),
+    RECAPTCHA_ENABLED=(bool, False),
+    RECAPTCHA_SECRET_KEY=(str, ''),
+    RECAPTCHA_SITE_KEY=(str, ''),
+    REDIS_URL=(str, 'redis://redis:6397/0'),
+    SENTRY_DSN=(str, None),
+    SESSION_COOKIE_SECURE=(bool, False),
+    SET_HSTS=(bool, False),
+    SSL_REDIRECT=(bool, False),
+    TARGET_DOMAINS=(list, []),
+    USE_S3=(bool, False),
+    USE_X_FORWARDED_FOR=(bool, False),
+    USE_X_FORWARDED_HOST=(bool, False),
+    WAGTAILLOCALIZE_PONTOON_GIT_CLONE_DIR=(str, ''),
+    WAGTAILLOCALIZE_PONTOON_GIT_URL=(str, ''),
+    X_FRAME_OPTIONS=(str, 'DENY'),
+    XSS_PROTECTION=(bool, True),
+)
+
+
+app = environ.Path(__file__) - 2
```

Comment:
is this... arithmetics on paths? O_O

Reply:
hahaha that was my reaction. but apparently it works. it's _magic_

Reply:
![skeptical face](https://media.giphy.com/media/600QaqOkDQnWo5Mau0/giphy.gif)

Reply:
I was curious, this is a `Path` class defined in django-environ: https://github.com/joke2k/django-environ/blob/develop/environ/environ.py#L735-L749